src/dataflow/components/nodes/controls/text-control.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from "react";
+import Rete from "rete";
+import "./text-control.sass";
+
+export class TextControl extends Rete.Control {
```

Comment:
This and the `NumControl` could potentially be collapsed into a single control.  Something work thinking about for the future.